src/Host/Client/Impl/Extensions/RawEvalResultExtensions.cs

```diff
@@ -7,16 +7,10 @@
 
 namespace Microsoft.R.Host.Client.Extensions {
     public static class RawEvalResultExtensions {
-        public static bool HasRawData(this REvaluationResult result) {
-            return result.Raw.Count > 0;
-        }
-
         public static void SaveRawDataToFile(this REvaluationResult result, string filepath) {
-            if (result.HasRawData()) {
+            if (result.RawResult != null) {
                 using (BinaryWriter writer = new BinaryWriter(File.OpenWrite(filepath))) {
-                    foreach (var data in result.Raw) {
-                        writer.Write(data);
-                    }
+                    writer.Write(result.RawResult);
```

Comment:
File.WriteAllBytes, I had used the binarywriter since there was a possibility that we send single file as multiple blobs.